### ISSUE TITLE
Allow array in `Query::$from` - preg_match() expects parameter 2 to be string, array given IN vendor/hiqdev/yii2-hiart/src/AbstractRequest.php:92

### ISSUE THREAD

**Issue by InAdavansP** [hiqdev/yii2-hiart#16]
Hi,

Here my config:

```
    'hiart' => [
        'class' => \hiqdev\hiart\rest\Connection::class,
        'requestClass' => \hiqdev\hiart\auto\Request::class,
        'baseUri' => 'https://site.com/api/v3/',
    ],
```

Using a model extending `hiqdev\hiart\ActiveRecord;`

```
<?php
namespace app\models\form;


use app\models\Analysis;
use hiqdev\hiart\ActiveRecord;
use Yii;


/**
 * Class NewRequestForm
 * @package app\models\form
 */
class NewRequestForm extends ActiveRecord
{
```

When i try to save the model : 

```
 if($model->save())
                    {
```

I have this error : 

```
yii\base\ErrorException: Undefined index: request in /dev/vendor/hiqdev/yii2-hiart/src/RequestErrorException.php:42
Stack trace:
#0 /dev/vendor/hiqdev/yii2-hiart/src/RequestErrorException.php(34): hiqdev\hiart\RequestErrorException->getErrorInfo()
#1 /dev/vendor/hiqdev/yii2-hiart/src/curl/Request.php(53): hiqdev\hiart\RequestErrorException->__construct()
#2 /dev/vendor/hiqdev/yii2-hiart/src/Command.php(120): hiqdev\hiart\curl\Request->send()
#3 /dev/vendor/hiqdev/yii2-hiart/src/Command.php(107): hiqdev\hiart\Command->send()
#4 /dev/vendor/hiqdev/yii2-hiart/src/ActiveRecord.php(261): hiqdev\hiart\Command->perform()
#5 /dev/vendor/hiqdev/yii2-hiart/src/ActiveRecord.php(249): hiqdev\hiart\ActiveRecord::perform()
#6 /dev/vendor/hiqdev/yii2-hiart/src/ActiveRecord.php(143): app\models\form\NewRequestForm->query()
#7 /dev/vendor/yiisoft/yii2/db/BaseActiveRecord.php(670): app\models\form\NewRequestForm->insert()
#8 /dev/controllers/InsuredController.php(197): app\models\form\NewRequestForm->save()
#9 /dev/vendor/yiisoft/yii2/base/InlineAction.php(57): app\controllers\InsuredController->actionNrfStepTreatment()
#10 /dev/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/dev/vendor/yiisoft/yii2/base/InlineAction.php:57}()
#11 /dev/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#12 /dev/vendor/yiisoft/yii2/base/Module.php(528): app\controllers\InsuredController->runAction()
#13 /dev/vendor/yiisoft/yii2/web/Application.php(103): yii\web\Application->runAction()
#14 /dev/vendor/yiisoft/yii2/base/Application.php(386): yii\web\Application->handleRequest()
#15 /dev/web/index.php(12): yii\web\Application->run()
#16 {main}
```

It come from 

> [vendor/hiqdev/yii2-hiart/src/rest/QueryBuilder.php]

   ```
 public function buildUri(Query $query)
    {
        return $query->from;
    }

```

called by : 
> vendor/hiqdev/yii2-hiart/src/AbstractRequest.php
```
protected function buildUri()
    {
        $this->uri = $this->builder->buildUri($this->query);
    }
```
We assume that  **$this->uri** have to be a string. But **$query->from** return the tables by an array.

So when the request has to be create preg_match fail:

```
public function createFullUri()
    {
        return ($this->isFullUri($this->uri) ? '' : $this->getDb()->getBaseUri()) . $this->uri;
    }

    public function isFullUri($uri)
    {
        return preg_match('/^https?:\\/\\//i', $uri);
    }
```

Is there something i did wrong?

**Comment by SilverFire** [comment]
Hello. Thank you for your detailed report.
Actually, we have never used a `from` property as an array in HiArt, so this case is not handled properly. Would you like to fix a pull request?

**Comment by InAdavansP** [comment]
Hi, thanks for your answer. 
I don't understand why the the `from` property is considered as a string in HiArt.
The `from` property extends from `yii\db\QueryInterface` who extends from `\yii\db\Query`.

The Yii2 documentation for this property says:
```
 /**
     * @var array the table(s) to be selected from. For example, `['user', 'post']`.
     * This is used to construct the FROM clause in a SQL statement.
     * @see from()
     */
    public $from;
```
In my project i'm never using more than one table in that array so i'm overriding this : 

```
public function buildUri(Query $query)
 {
     return $query->from;
 }
```

====>

```
public function buildUri(Query $query)
 {
     return $query->from[0];
 }
```
but i think it's not a stable option.

**Comment by InAdavansP** [comment]
By the way this issue for the `Basic REST`  class has the connected issue for the `Guzzle` class : 

> #14 parse_url() expects parameter 1 to be string, array given

**Comment by hiqsol** [comment]
Fixed with PR #20
